src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -574,6 +574,8 @@ public void AttachToDependencyGraph(DependencyAnalysisFramework.DependencyAnalyz
             graph.AddRoot(ReadyToRunHeader, "ReadyToRunHeader is always generated");
             graph.AddRoot(new ModulesSectionNode(Target), "ModulesSection is always generated");
 
+            AddWellKnownTypes(graph);
```

Comment:
Why doesn't this fall out of the dependency graph? Ideally we shouldn't root anything because if the dependency graph doesn't need these, they're garbage.

If this is _really_ needed, it should at least go to the `CppCodegenNodeFactory` class. Feel free to make `AttachToDependencyGraph` virtual so that you can override it there.